src/org/javarosa/core/model/SelectChoice.java

```diff
@@ -132,4 +133,27 @@ public void setTextID(String textID) {
         this.textID = textID;
     }
 
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+        SelectChoice otherChoice = (SelectChoice) other;
+
+        return Objects.equals(labelInnerText, otherChoice.labelInnerText)
+            && Objects.equals(textID, otherChoice.textID)
+            && isLocalizable == otherChoice.isLocalizable
+            && Objects.equals(value, otherChoice.value)
+            && index == otherChoice.index
+            && copyNode == otherChoice.copyNode;
```

Comment:
I always use IntelliJ's automatic generator of equals and hash methods. The only difference I see is that IntelliJ suggests `Objects.equals(copyNode, otherChoice.copyNode)`. Are you sure you want to compare the `copyNode` by reference and not by value?

Reply:
I made the change and pushed it but I don't even know, really. `TreeElement`s aren't meaningfully comparable because again, they're mutable.

Reply:
I think it's OK: using Object.equals is safer regarding null values (not super important in this particular case), and equality rules are what we need unless we implement `TreeElement.equals()` (which should happen only if we make it immutable or implement an `equals` method we're happy about)